src/include/mpir_hw_topo.h

```diff
@@ -0,0 +1,189 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2019 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ */
+
+#ifndef MPIR_HW_TOPO_H_INCLUDED
+#define MPIR_HW_TOPO_H_INCLUDED
+
+/*
+ * Tracks hwloc_obj_type_t do not change ordering
+ */
+typedef enum {
+    MPIR_HWLOC_OBJ_MACHINE,
+    MPIR_HWLOC_OBJ_PACKAGE,
+    MPIR_HWLOC_OBJ_CORE,
+    MPIR_HWLOC_OBJ_PU,
+    MPIR_HWLOC_OBJ_L1CACHE,
+    MPIR_HWLOC_OBJ_L2CACHE,
+    MPIR_HWLOC_OBJ_L3CACHE,
+    MPIR_HWLOC_OBJ_L4CACHE,
+    MPIR_HWLOC_OBJ_L5CACHE,
+    MPIR_HWLOC_OBJ_L1ICACHE,
+    MPIR_HWLOC_OBJ_L2ICACHE,
+    MPIR_HWLOC_OBJ_L3ICACHE,
+    MPIR_HWLOC_OBJ_GROUP,
+    MPIR_HWLOC_OBJ_NUMANODE,
+    MPIR_HWLOC_OBJ_BRIDGE,
+    MPIR_HWLOC_OBJ_PCI_DEVICE,
+    MPIR_HWLOC_OBJ_OS_DEVICE,
+    MPIR_HWLOC_OBJ_MISC,
+    MPIR_HWLOC_OBJ_TYPE_MAX,
+    MPIR_MPICH_OBJ_DDR,         /* MPICH object for DDR memory */
+    MPIR_MPICH_OBJ_MCDRAM,      /* MPICH object for MCDRAM memory */
+    MPIR_MPICH_OBJ_DEFAULT = MPIR_MPICH_OBJ_DDR
+} MPIR_Hwloc_obj_type;
+
+/*
+ * Tracks hwloc_membind_policy_t do not change
+ */
+typedef enum {
+    MPIR_HWLOC_MEMBIND_DEFAULT = 0,
+    MPIR_HWLOC_MEMBIND_FIRSTTOUCH = 1,
+    MPIR_HWLOC_MEMBIND_BIND = 2,
+    MPIR_HWLOC_MEMBIND_INTERLEAVE = 3,
+    MPIR_HWLOC_MEMBIND_NEXTTOUCH = 4,
+    MPIR_HWLOC_MEMBIND_MIXED = -1
+} MPIR_Hwloc_membind_policy;
```

Comment:
How about `MPIR_MEMBIND_POLICY__*`?  Also, by `default`, do you mean system-specified?